src/odin/maneuversbuilder.cc

```diff
@@ -3939,5 +3946,40 @@ void ManeuversBuilder::CollapseMergeManeuvers(std::list<Maneuver>& maneuvers) {
   }
 }
 
+void ManeuversBuilder::AddLandmarksFromTripLegToManeuvers(std::list<Maneuver>& maneuvers) {
+  std::vector<RouteLandmark> landmarks{};
+  for (auto man = maneuvers.begin(); man != maneuvers.end(); ++man) {
+    // set landmarks correlated with edges in the previous maneuver to the current maneuver
+    if (!landmarks.empty()) {
+      man->set_landmarks(landmarks);
+    }
+    landmarks.clear();
+
+    // accumulate landmarks in the current maneuver
+    double distance_from_begin_to_curr_edge = 0; // distance from the begin point of the whole
+                                                 // manerver to the begin point of the current edge
+    double maneuver_total_distance =
+        man->length() * kConvertKmtoMeters; // total distance of the maneuver in meters
```

Comment:
this is in constants.h already
```suggestion
        man->length() * kMetersPerKm; // total distance of the maneuver in meters
```